packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -646,43 +678,157 @@ class FocusNode with DiagnosticableTreeMixin, ChangeNotifier {
+  ///               return SizedBox(
+  ///                 width: 200,
+  ///                 child: Padding(
+  ///                   padding: const EdgeInsets.all(8.0),
+  ///                   child: TextField(
+  ///                     decoration: InputDecoration(border: OutlineInputBorder()),
+  ///                   ),
+  ///                 ),
+  ///               );
+  ///             }),
+  ///           ),
+  ///           Row(
+  ///             mainAxisAlignment: MainAxisAlignment.spaceAround,
+  ///             children: <Widget>[
+  ///               ...List<Widget>.generate(UnfocusDisposition.values.length,
+  ///                   (int index) {
+  ///                 return Row(
+  ///                   mainAxisSize: MainAxisSize.min,
+  ///                   children: <Widget>[
+  ///                     Radio<UnfocusDisposition>(
+  ///                       groupValue: disposition,
+  ///                       onChanged: (UnfocusDisposition value) {
+  ///                         setState(() {
+  ///                           disposition = value;
+  ///                         });
+  ///                       },
+  ///                       value: UnfocusDisposition.values[index],
+  ///                     ),
+  ///                     Text(describeEnum(UnfocusDisposition.values[index])),
+  ///                   ],
+  ///                 );
+  ///               }),
+  ///               OutlineButton(
+  ///                 child: const Text('UNFOCUS'),
+  ///                 onPressed: () {
+  ///                   setState(() {
+  ///                     primaryFocus.unfocus(disposition: disposition);
+  ///                   });
+  ///                 },
+  ///               ),
+  ///             ],
+  ///           ),
+  ///         ],
+  ///       ),
+  ///     ),
+  ///   );
+  /// }
+  /// ```
+  /// {@end-tool}
+  void unfocus({
+    UnfocusDisposition disposition = UnfocusDisposition.scope,
+  }) {
+    assert(disposition != null);
     if (!hasFocus && (_manager == null || _manager._markedForFocus != this)) {
       return;
     }
-    if (!hasPrimaryFocus) {
-      // If we are in the focus chain, but not the primary focus, then unfocus
-      // the primary instead.
-      _manager?.primaryFocus?.unfocus(focusPrevious: focusPrevious);
+    FocusScopeNode scope = enclosingScope;
+    if (scope == null) {
+      // Attempting to unfocus the root scope isn't allowed.
+      return;
     }
-    _manager?._markUnfocused(this);
-    final FocusScopeNode scope = enclosingScope;
-    if (scope != null) {
-      scope._focusedChildren.remove(this);
-      if (focusPrevious) {
-        scope._doRequestFocus();
-      }
+    switch (disposition) {
+      case UnfocusDisposition.scope:
+        // Focus the nearest scope without focusing its "focused child" or
+        // modifying the first focus for itself or any descendant scopes, except
+        // for removing the currently focused node from its _focusedChildren,
```

Comment:
line 810 clears all the focus nodes. It is contradicting with the comment

Reply:
Thanks, I removed the comment, because it no longer does that.